addon/helpers/date-diff.js

```diff
@@ -21,6 +23,17 @@ import normalizeDate from '../utils/normalize-date';
 export default helper(function dateDiff([dateA, dateB], { precision = 'days', inputFormat }) {
   if (!dateA || !dateB) return null;
 
+  deprecate(
+    `{{date-diff date args should be matching type, you passed '${typeOf(dateA)}' and '${typeOf(
+      dateB
+    )}'`,
+    typeOf(dateA) === typeOf(dateB),
+    {
+      id: 'date-diff/avoid-mixed-date-types',
+      until: '4.0.0',
```

Comment:
```suggestion
      until: '1.0.0',
```
I think this `until` references our addon correct?

Reply:
according to the API docs.. `until` is supposed to be the Ember version.. maybe I read them too literally?

Reply:
Hmm... you are correct (reference for future time travelers: https://github.com/emberjs/ember.js/blob/79f130f779a58969b98d079acf7d0e83c81dae63/packages/%40ember/debug/lib/deprecate.ts#L193-L194).

What happens if you change this to 1.0.0? Do the deprecation warnings cause issues?